### ISSUE TITLE
If-present-then expressions

### ISSUE THREAD

**Issue by tbateson** [horseshoe/horseshoe#130]
Feature request for if-then scoping rules. This a syntactic sugar that
```handlebars
{{ myBool ? myValue : null }}
```
is equivalent to
```handlebars
{{ myBool => myValue }}
```

This operator is essentially the opposite of the null-coalescent operator where the expression is only evaluated when the expression is null/false.

Note that the first expression need not be a boolean, but could be an coercible value such as shown below:
```handlebars
{{ stringValue := "123" }}
{{ stringSecond := "abc" }}
{{# stringValue => stringSecond }}{{.}}{{! The resulting render is "abc" }}{{/}}
{{ stringValue => stringSecond }}{{! Same output as above }}
{{ thisValueIsNull => stringValue }}{{! Output is null/empty }}
```

Additionally, I would like for this feature to support annotations as well:
```handlebars
{{# myBool => @StdOut }}
This prints to stdout, but only if myBool is true.
{{/}}
```
This scoping feature especially for annotations would be incredibly useful.

**Comment by nicklauslittle** [comment]
This is partially addressed by #131.

**Comment by nicklauslittle** [comment]
The `=>` syntax cannot be used, as it is reserved for named expressions. However, I believe these are both addressed. The first example (`{{ myBool => myValue }}`) can use the Inverted Elvis operator: `{{ myBool !: myValue }}`. This operator fulfills the coercible requirements listed as well.

The annotation support has been improved so that annotations can be invoked inside expressions. This allows them to be used with the Inverted Elvis operator as well: `{{# myBool !: @StdOut }}`. This also works for annotations that take arguments (e.g., `{{# renderToFile !: @File(['name': 'MyFile.txt', 'overwrite': true]) }}`). Here is an example:

```horseshoe
{{# myBool !: @StdOut }}
This prints to stdout, but only if myBool is true.
{{^# @StdErr }}
This prints to stderr if myBool is false
{{/ @StdErr }}
```

**Comment by nicklauslittle** [comment]
Closed by #153. Implemented in Horseshoe 2.0.
